### ISSUE TITLE
Firebase Database error on Import in Editor Unity

### ISSUE THREAD

**Issue by studentutu** [firebase/quickstart-unity#236]
Errorrs:
Failed to rename asset Assets/Plugins/x86_64/Firebase/libFirebaseCppApp-5.4.0.so to Assets\Plugins\x86_64\Firebase\libFirebaseCppApp-5.4.0.so (The source asset could not be found)

Failed to rename asset Assets/Plugins/x86_64/Firebase/libFirebaseCppAuth.so to Assets\Plugins\x86_64\Firebase\libFirebaseCppAuth.so (The source asset could not be found)

Failed to rename asset Assets/Plugins/x86_64/Firebase/libFirebaseCppDatabase.so to Assets\Plugins\x86_64\Firebase\libFirebaseCppDatabase.so (The source asset could not be found)

Steps to reproduce:
1. Unity 2018.14f1,  .Net 4.x, Android
2. Install Latest Google Play Games plugins for Unity
3. Install Google lastest Admob Unity Plugin
(Try to build just in case anything else breaks -> Everything should be okay and App is working)
4. Install Latest  Firebase Database plugin (5.4.0)

You will get an errors and all .so files will be deleted from the Assets/Plugins/x86_64/Firebase/

As a result, Build succeeds but the application crashes on Lunch - there no .so files.

**Comment by stewartmiles** [comment]
@studentutu the .so files you've referenced are used on Linux .  What platform are you launching on?

It sounds like this issue is rearing it's head: https://github.com/googlesamples/unity-jar-resolver/issues/159

**Comment by studentutu** [comment]
@stewartmiles   I have made another build - 
 1. Unity 2018.14f1, .Net 3.x, Android, build on Windows 10
2. Install Latest Google Play Games plugins for Unity
3. Install Latest Firebase Database plugin (5.4.0)  (No Errors occurs on Import)
(Try to build just in case anything else breaks -> Everything should be okay and App is working)
4.  Install Google latest Admob Unity Plugin  (No Errors occurs on Import)

Now :Try to build just in case anything else breaks but App crashes right on Launch


Long Story Short:
Admob Plugin somehow breaks the App and dependencies. Not only we have https://github.com/googlesamples/unity-jar-resolver/issues/159, but also new Issue - Latest Admob Unity plugin breaks the  app

**Comment by studentutu** [comment]
https://github.com/firebase/quickstart-unity/issues/236#issuecomment-434787204, it was fixed by simply adding the app id to the manifest, but still there is a bug with .Net 4.x https://github.com/firebase/quickstart-unity/issues/236#issue-376058127

**Comment by MGGDev** [comment]
I was getting something similar.
My project has AdMob, Facebook SDK (without importing their Resolver, as it causes problems) and Playfab. Then I import Messaging, Storage, and Authenticate (imported in this order, one by one).
I didn't import the Resolver in the Storage and Auth at first, but that caused the error below. So I skipped the Storage resolver, and re-imported Auth with its Resolver. Then hit Built. Success!

Here is the error:
Failed to rename asset Assets/Plugins/x86_64/Firebase/libFirebaseCppAuth.so to Assets\Plugins\x86_64\Firebase\libFirebaseCppAuth.so (The source asset could not be found)
UnityEngine.Debug:LogError(Object)
Google.Logger:Log(String, LogLevel) (at Z:/tmp/tmp.iUQPObMiPM/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/Logger.cs:91)
Google.VersionHandlerImpl:Log(String, Boolean, LogLevel) (at Z:/tmp/tmp.iUQPObMiPM/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/VersionHandlerImpl.cs:1658)
Google.FileMetadata:RenameAsset(String) (at Z:/tmp/tmp.iUQPObMiPM/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/VersionHandlerImpl.cs:573)
Google.LinuxLibraryRenamer:RenameLibraries() (at Z:/tmp/tmp.iUQPObMiPM/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/VersionHandlerImpl.cs:1403)
Google.VersionHandlerImpl:UpdateVersionedAssets(Boolean) (at Z:/tmp/tmp.iUQPObMiPM/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/VersionHandlerImpl.cs:1749)
Google.VersionHandlerImpl:UpdateVersionedAssetsOnUpdate() (at Z:/tmp/tmp.iUQPObMiPM/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/VersionHandlerImpl.cs:1463)
Google.VersionHandlerImpl:<VersionHandlerImpl>m__0() (at Z:/tmp/tmp.iUQPObMiPM/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/VersionHandlerImpl.cs:1459)
Google.RunOnMainThread:ExecuteNext() (at Z:/tmp/tmp.iUQPObMiPM/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:363)
Google.RunOnMainThread:<ExecuteAll>m__1() (at Z:/tmp/tmp.iUQPObMiPM/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:394)
Google.RunOnMainThread:RunAction(Action) (at Z:/tmp/tmp.iUQPObMiPM/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:234)
Google.RunOnMainThread:ExecuteAll() (at Z:/tmp/tmp.iUQPObMiPM/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:392)
UnityEditor.EditorApplication:Internal_CallUpdateFunctions()

**Comment by MGGDev** [comment]
I am facing the same problem again, but I can't fix it.
Is there any work arounds? I have tried Firebase Admob SDK and it fails as well.

**Comment by stewartmiles** [comment]
When in doubt, always upgrade the resolver.  The latest release is always available on https://github.com/googlesamples/unity-jar-resolver .

@MGGDev I recommend removing `Assets/PlayServicesResolver` and importing the latest version of the resolver plugin.

**Comment by MGGDev** [comment]
Thanks a lot @stewartmiles !
I used the latest resolver instead of those which are bundled with the plugins, but I got an error:
> Argument cannot be null. Parameter name: path1
It said something about Android SDK. So I updated it using Android Studio (used a path name without spaces in it), and set the new path in Unity->Edit->Preferences (along with using Unity's OpenJDK).

Everything works now. Thanks again :)
